forch/local_state_collector.py

```diff
@@ -149,40 +149,49 @@ def _extract_process_state(self, proc_name, proc_count, proc_list):
             proc_map['create_time_change_count'] = change_count
             proc_map['create_time_last_change'] = self._current_time
 
-        error = self._aggregate_process_stats(proc_map, proc_list)
+        try:
+            self._aggregate_process_stats(proc_map, proc_list)
+        except Exception as e:
+            return None, f'Error in extracting info for process {proc_name}: {e}'
 
-        if error:
-            return None, error
+        target_proc_config = self._target_procs[proc_name]
+        if target_proc_config.cpu_percent_threshold:
+            if proc_map['cpu_percent'] > target_proc_config.cpu_percent_threshold:
+                error = f'CPU percent of process {proc_name} is {proc_map["cpu_percent"]}, '
+                error += f'exceeding threashold {target_proc_config.cpu_percent_threshold}'
```

Comment:
Is this done periodically or on demand from the API?

Also, does it log anything in the case where it happens, or only return an error?  The logging would likely be very useful since it would then correlate in the log file with any triggering activity.

Reply:
This is periodically, along with the existing process polling.

Added a logging at 'error' level.